### PR TITLE
Handle newHeads websocket subscriptions

### PR DESCRIPTION
**Pull request by LukasGasior1**

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala [hunk 1]
```diff
@@ -566,6 +569,11 @@ class RegularSync(
   private def notDownloading(): Boolean =
     headersQueue.isEmpty && waitingForActor.isEmpty && !resolvingBranches
 
+  private def publishEvents(newBlocks: Seq[Block]): Unit = {
```
Comment (rtkaczyk) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala]: Don't we need this in `handleBlocks` as well?

File: src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala [hunk 2]
```diff
@@ -566,6 +569,11 @@ class RegularSync(
...
+    newBlocks.foreach { newBlock =>
+      context.system.eventStream.publish(NewHead(newBlock))
```
Comment (rtkaczyk) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala]: Do we need to publish it for all blocks and not just the latest one?

**Comment by rtkaczyk** [comment]
Could you provide more information about this (links)? How do I test it using web3?

File: src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala (hunk 2 above)
Comment (LukasGasior1) [reply src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala]: For all blocks

File: src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala (hunk 1 above)
Comment (LukasGasior1) [reply src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala]: true, we do

**Comment by LukasGasior1** [comment]
@rtkaczyk https://web3js.readthedocs.io/en/1.0/web3-eth-subscribe.html#subscribe-newblockheaders this uses the `newHeads` subscription

**Comment by rtkaczyk** [comment]
Is it really called `newHeads`? The doc only mentions `newBlockHeaders`

**Comment by LukasGasior1** [comment]
newHeads is the underlying json rpc call (https://github.com/ethereum/go-ethereum/wiki/RPC-PUB-SUB#newheads, newBlockHeaders is web3 naming

**Pull request merged**